odoo/addons/test_http/tests/test_static.py

```diff
@@ -254,6 +255,101 @@ def test_static15_public_user_image(self):
 
 
+@tagged('post_install', '-at_install')
+class TestHttpStaticLogo(TestHttpStaticCommon):
+    @staticmethod
+    def img_data_to_web_data(img_base_64):
+        return image_process(img_base_64, size=(180, 0))
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        ResCompany = cls.env['res.company']
+        cls.default_logo_data = TestHttpStaticLogo.img_data_to_web_data(base64.b64decode(ResCompany._get_logo()))
+        cls.gizeh_data_b64 = base64.encodebytes(cls.gizeh_data)
+        cls.logo_gizeh_data = TestHttpStaticLogo.img_data_to_web_data(cls.gizeh_data)
+        with file_open('web/static/img/nologo.png', 'rb') as file:
+            cls.logo_no_logo_data = file.read()
+        cls.logo_no_logo_data_b64 = base64.encodebytes(cls.logo_no_logo_data)
+        cls.headers_default_logo = {
+            'Content-Length': f'{len(cls.default_logo_data)}',
+            'Content-Type': 'image/png',
+            'Content-Disposition': 'inline; filename=logo.png'
+        }
+        cls.headers_logo_gizeh = {
+            'Content-Length': f'{len(cls.logo_gizeh_data)}',
+            'Content-Type': 'image/png',
+            'Content-Disposition': 'inline; filename=logo.png'
+        }
+        cls.headers_logo_no_logo = {
+            'Content-Length': f'{len(cls.logo_no_logo_data)}',
+            'Content-Type': 'image/png',
+            'Content-Disposition': 'inline; filename=nologo.png'
+        }
+        companies = ResCompany.search([]) | ResCompany.create(
+            {
+                'name': 'Company 2',
+                'email': 'company.2@test.example.com',
+                'country_id': cls.env.ref('base.fr').id,
+            }
+        )
+        cls.company1, cls.company2 = companies
+        cls.user_company1, cls.user_company2 = [
+            new_test_user(cls.env, f'user_{company.id}', company_id=company.id, password='test')
+            for company in companies]
+
+    def assertDownloadLogo(self, headers, img_data, user=None, company=None, all_endpoints=False):
+        """Assert that the logo endpoint returns the right image and headers.
+
+        :param dict headers: expected headers
+        :param bytes img_data: expected image data
+        :param user: optional user, if set the check will be done while being authenticated with that user
+        :param company: optional company, the company will be appended in the URL parameters
+        :param all_endpoints: whether to test all endpoint or only the main one
+        """
+        url_suffix = f'?company={company.id}' if company else ''
+        if user:
+            self.authenticate(user.login, 'test')
+        else:
+            self.authenticate(None, None)
+        self.assertDownload(f'/logo.png{url_suffix}', 200, headers, img_data)
+        if all_endpoints:
+            self.assertDownload(f'/logo{url_suffix}', 200, headers, img_data)
+            self.assertDownload(f'/web/binary/company_logo{url_suffix}', 200, headers, img_data)
```

Comment:
This bit is not necessary, you don't need to test that multiple routes can serve a same controller endpoint, that's covered by another test already :)

Reply:
I have removed the test of the extra endpoints (my goal was to ensure the endpoints were not removed in the future); now only "/logo.png" is tested.